### PR TITLE
fix: Use of SQL `ORDER BY` should not cause reordering of `SELECT` cols

### PR DESCRIPTION
**Pull request by alexander-beedie**
It was possible for use of an `ORDER BY` clause to indirectly cause reordering of `SELECT` cols when using the SQL interface. Fixed, and added some test coverage.

### CODE REVIEW

File: crates/polars-sql/src/context.rs [hunk 1]
```diff
@@ -535,40 +535,44 @@ impl SQLContext {
             if query.order_by.is_empty() {
                 lf.select(projections)
             } else if !contains_wildcard {
+                let mut retained_names = Vec::with_capacity(projections.len());
```
Comment (ritchie46) [inline crates/polars-sql/src/context.rs:538]: We can use `PlIndexSet` if order is important but still want fast lookups and removals.

Comment (alexander-beedie) [reply crates/polars-sql/src/context.rs:538]: In this case we're just building a Vec to pass to `select`

Comment (alexander-beedie) [reply crates/polars-sql/src/context.rs]: Yup, `exclude` benefits from fast removes - modified accordingly!✌️

**Comment by alexander-beedie** [comment]
Updated/rebased

File: crates/polars-sql/src/context.rs [hunk 2]
```diff
@@ -557,18 +556,23 @@ impl SQLContext {
...
                     },
-                    _ => {},
                 });
-
                 lf = lf.with_columns(projections);
                 lf = self.process_order_by(lf, &query.order_by)?;
-
-                column_names.retain(|&name| !retained_names.contains(name));
-                lf.drop(column_names)
+                lf.select(
```
Comment (ritchie46) [inline crates/polars-sql/src/context.rs:570]: You can call `as_slice` on retained_names.

**Pull request merged**